cmd/kubelet/app/options/options.go

```diff
@@ -272,4 +274,6 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.DurationVar(&s.EvictionPressureTransitionPeriod.Duration, "eviction-pressure-transition-period", s.EvictionPressureTransitionPeriod.Duration, "Duration for which the kubelet has to wait before transitioning out of an eviction pressure condition.")
 	fs.Int32Var(&s.EvictionMaxPodGracePeriod, "eviction-max-pod-grace-period", s.EvictionMaxPodGracePeriod, "Maximum allowed grace period (in seconds) to use when terminating pods in response to a soft eviction threshold being met.  If negative, defer to pod specified value.")
 	fs.Int32Var(&s.PodsPerCore, "pods-per-core", s.PodsPerCore, "Number of Pods per core that can run on this Kubelet. The total number of Pods on this Kubelet cannot exceed max-pods, so max-pods will be used if this calculation results in a larger number of Pods allowed on the Kubelet. A value of 0 disables this limit.")
+	fs.StringVar(&s.KernelTunableBehavior, "kernel-tunable-behaviour", s.KernelTunableBehavior, "Default kubelet behaviour for kernel tunning. Valid values: warning, modify, error")
```

Comment:
`warn` is dubious. It will make debugging hard. Unless one understands most
of the kubelet functionalities that depend on these kernel configs, once
cannot safely set `warn`. Given the current state of the project, I'd
prefer not supporting `warn`.

On Wed, Jun 22, 2016 at 11:44 AM, Derek Carr <notifications@github.com>
wrote:

> In cmd/kubelet/app/options/options.go
> <https://github.com/kubernetes/kubernetes/pull/27874#discussion_r68111338>
> :
>
> > @@ -272,4 +274,6 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
> >  	fs.DurationVar(&s.EvictionPressureTransitionPeriod.Duration, "eviction-pressure-transition-period", s.EvictionPressureTransitionPeriod.Duration, "Duration for which the kubelet has to wait before transitioning out of an eviction pressure condition.")
> >  	fs.Int32Var(&s.EvictionMaxPodGracePeriod, "eviction-max-pod-grace-period", s.EvictionMaxPodGracePeriod, "Maximum allowed grace period (in seconds) to use when terminating pods in response to a soft eviction threshold being met.  If negative, defer to pod specified value.")
> >  	fs.Int32Var(&s.PodsPerCore, "pods-per-core", s.PodsPerCore, "Number of Pods per core that can run on this Kubelet. The total number of Pods on this Kubelet cannot exceed max-pods, so max-pods will be used if this calculation results in a larger number of Pods allowed on the Kubelet. A value of 0 disables this limit.")
> > +	fs.StringVar(&s.KernelTunableBehavior, "kernel-tunable-behaviour", s.KernelTunableBehavior, "Default kubelet behaviour for kernel tunning. Valid values: warning, modify, error")
>
> I had thought a few scenarios:
>
>    - warn
>       - useful when operators have a different opinion than our default
>       opinion (it exists, or could exist if we set others)
>       - useful when doing hack/local-up-cluster.sh on your dev machine
>       (and i dont need the tunables to change)
>    - error
>       - useful when operators want to pre-configure kernel tunables and
>       want no suprises (this is our ops case)
>    - modify
>       - useful when operators just want defaults (this is the default
>       good experience)
>
> I think a bool that fails is not flexible enough.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27874/files/53c0488a330dec757deb599263219c606333f3f8#r68111338>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGvIKJ2YJTXXS03n-qoC4HqNCTpK85QXks5qOYKKgaJpZM4I713Y>
> .
>